clang/lib/CIR/Dialect/Transforms/LoweringPrepare.cpp

```diff
@@ -123,13 +123,16 @@ struct LoweringPreparePass : public LoweringPrepareBase<LoweringPreparePass> {
   /// CUDA related
   /// ------------
 
-  // Maps CUDA device stub name to kernel name.
-  llvm::DenseMap<llvm::StringRef, std::string> cudaKernelMap;
+  // Maps CUDA kernel name to device stub function.
+  std::unordered_map<std::string, FuncOp> cudaKernelMap;
```

Comment:
Any specific reason for the data structure change? Have you considered `llvm::StringMap`?